### PR TITLE
Results controller implementation

### PR DESCRIPTION
**Pull request by basiliskus**
# Results controller implementation

- Add necessary objects for implementing Result Controller
- Add ResultController Implementation
- Add ResultResponse Implementation
- Implement handleResults in the EtorDomainRegistration and register the endpoint.
  - Register ResultController, ResultResponse, and SendResultUseCase.
- Add test case scenarios for coverage

## Issue

#615 

## Checklist

- [ ] I have added tests to cover my changes
- [ ] I have added logging where useful (with appropriate log level)
- [ ] I have added JavaDocs where required
- [ ] I have updated the documentation accordingly

**Note**: You may remove items that are not applicable

### CODE REVIEW

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java [hunk 1]
```diff
@@ -231,6 +242,34 @@ DomainResponse handleMetadata(DomainRequest request) {
...
+            receivedSubmissionId = null;
+            logger.logError("Missing required header or empty: RecordId");
+        }
+
+        String errorMessage = "";
+        try {
+            results = resultController.parseResults(request);
+            sendResultUseCase.convertAndSend(results, receivedSubmissionId);
+        } catch (FhirParseException e) {
+            errorMessage = "Unable to parse result request";
```
Comment (basiliskus) [inline etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java:259]: If the error message string is only used in place, I don't think a variable would be needed. Same for the one in the other catch

**Review by basiliskus — COMMENTED** [review]
Looks great Sam! I just left one comment with a minor suggestion. Also, since this PR is adding the results endpoint, can you add an end-to-end test for it? You can check `e2e/src/test/groovy/gov/hhs/cdc/trustedintermediary/e2e/OrderTest.groovy` as an example

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java [hunk 2]
```diff
@@ -231,6 +241,31 @@ DomainResponse handleMetadata(DomainRequest request) {
...
+        } catch (FhirParseException e) {
+            logger.logError("Unable to parse result request", e);
+            return domainResponseHelper.constructErrorResponse(400, e);
+        } catch (UnableToSendMessageException e) {
+            logger.logError("Unable to send result", e);
+            return domainResponseHelper.constructErrorResponse(400, e);
+        }
+
+        ResultResponse resultResponse = new ResultResponse(results);
+        logger.logInfo(request.getHeaders().toString());
```
Comment (basiliskus) [inline etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java:265]: Is this logging needed here?

**Review by basiliskus — COMMENTED** [review]
Looks good! Just let one minor comment

**Review by saquino0827 — APPROVED** [review]

**Pull request merged**